compiler/rustc_hir/src/hir.rs

```diff
@@ -425,6 +425,7 @@ pub enum TraitBoundModifier {
     None,
     Negative,
     Maybe,
+    Const,
     MaybeConst,
 }
```

Comment:
there is a doc above saying that this is only used for `?Sized`, which is not the case, probably good to add documentation to each variant too, like `!Trait`, `?Trait`, `const Trait`, and `~const Trait` for each variant.

Reply:
Ye, that doc comment was severely outdated. I've already removed it in #119163 (gonna rebase my PR once I get home) but I can also add doc comments to each variant 👍